jena-extras/jena-serviceenhancer/src/test/java/org/apache/jena/sparql/service/enhancer/impl/TestServiceEnhancerMisc.java

```diff
@@ -462,4 +471,55 @@ public void testNormalization01() {
         int actualRowCount = AbstractTestServiceEnhancerResultSetLimits.testWithCleanCaches(dataset, queryStr, 1000);
         Assert.assertEquals(4, actualRowCount);
     }
+
+    /**
+     * Test for <a href="https://github.com/apache/jena/issues/1688">JENA-1688</a>.
+     * <p>
+     * This test checks that building an overall result set from a bulk request
+     * that involves contributions of cached empty result sets works as expected.
+     * Without the fix corresponding to this issues this test fails.
+     *
+     * @implNote This test case makes use of a dataset where only a few resources have labels.
+     * The test query caches the labels of all resources which means that most cache entries
+     * have empty results.
+     */
+    @Test
+    public void testEmptyResultSetContribution() {
+        String dataStr = String.join(StandardSystemProperty.LINE_SEPARATOR.value(),
```

Comment:
Not critical but why not "\n"? SPARQL is a web language.

Reply:
Isn't it the case that on Windows or Mac a plain "\n" could lead to incorrect formatting when `System.out.println`-ing?
If you prefer "\n" then I am also fine with it.